### PR TITLE
Apply manual coda codes

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Review https://github.com/AfricasVoices/Project-ICRAF/pull/7 first

### CODE REVIEW

File: src/apply_manual_codes.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                coda_input_path = path.join(coda_input_dir, plan.coda_filename)
+                if path.exists(coda_input_path):
+                    f = open(coda_input_path, "r")
+                TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable(
+                    user, data, plan.id_field, {plan.coded_field: plan.code_scheme}, f)
+            finally:
+                if f is not None:
+                    f.close()
+        
+        for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
```
Comment (as2388) [inline src/apply_manual_codes.py]: This looks like the same block of code as the one above. If it is the same, consider `for plan in PipelineConfiguration.DEMOGS_CODING_PLANS + FOLLOW_UP_CODING_PLANS`. If it isn't, please add a comment that draws attention to the difference

File: src/apply_manual_codes.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                    missing_dict[plan.coded_field] = na_label.to_dict()
+                elif td[plan.raw_field] == "":
+                    nc_label = CleaningUtils.make_label_from_cleaner_code(
+                        plan.code_scheme, plan.code_scheme.get_code_with_control_code(Codes.NOT_CODED),
+                        Metadata.get_call_location()
+                    )
+                    missing_dict[plan.coded_field] = nc_label.to_dict()
+            td.append_data(missing_dict, Metadata(user, Metadata.get_call_location(), time.time()))
+        
+            for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
```
Comment (as2388) [inline src/apply_manual_codes.py]: ditto

File: src/apply_manual_codes.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                if f is not None:
+                    f.close()
+        
+        for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+            f = None
+            try:
+                coda_input_path = path.join(coda_input_dir, plan.coda_filename)
+                if path.exists(coda_input_path):
+                    f = open(coda_input_path, "r")
+                TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable(
```
Comment (as2388) [inline src/apply_manual_codes.py]: Actually shouldn't you be treating this like a radio question? Given that your follow-up surveys need multi-coding shouldn't this be using the multi-coded importer?

**Review by as2388 — CHANGES_REQUESTED** [review]

File: src/apply_manual_codes.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+                    if plan.binary_code_scheme is not None:
+                        na_label = CleaningUtils.make_label_from_cleaner_code(
+                            plan.binary_code_scheme, plan.binary_code_scheme.get_code_with_control_code(Codes.TRUE_MISSING),
+                            Metadata.get_call_location()
+                        )
+                        missing_dict[plan.binary_coded_field] = na_label.to_dict()
+
+            td.append_data(missing_dict, Metadata(user, Metadata.get_call_location(), time.time()))
+
```
Comment (as2388) [inline src/apply_manual_codes.py:83]: I've realised you do need another block of code in here that sets 'NC's on the RQAs and Follow-Ups where the raw text is "". You should be able to adapt the code you used for doing this with the demogs below.

**Review by as2388 — CHANGES_REQUESTED** [review]

File: src/apply_manual_codes.py (hunk 1 above)
Comment (IsaackMwenda) [reply src/apply_manual_codes.py]: Updated to Multi code!

File: src/apply_manual_codes.py (hunk 2 above)
Comment (IsaackMwenda) [reply src/apply_manual_codes.py]: Updated to Multi code

File: src/apply_manual_codes.py (hunk 3 above)
Comment (IsaackMwenda) [reply src/apply_manual_codes.py]: Updated to Multi coded importer thanks

File: src/apply_manual_codes.py (hunk 4 above)
Comment (IsaackMwenda) [reply src/apply_manual_codes.py:83]: Updated PTAL @as2388

Comment (as2388) [reply src/apply_manual_codes.py:83]: You need to set the binary code scheme to NC too

**Review by as2388 — CHANGES_REQUESTED** [review]

**Pull request merged**